### DISCUSSION TITLE
Possible values for osm_value

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#674] [Q&A] [answered]
Hello everyone. The field `osm_value` in Elasticsearch index used by Photon is value of the main tag of the OSM object. How is this value extracted from raw OpenStreetMap data? I tried to read source code in Nominatim but didn't manage to find something useful.

Thank you! 🙏

**Comment by mtmail** [comment, marked as answer]
It's the the same as the `type` column in Nominatim's `placex` table. `photon/website/import.py` has the relevant logic.

**Reply by xinyuluo** [reply]
@mtmail thanks for your answer! I'm more curious how this value is extracted from OpenStreetMap data at all. For instance, one possible value for `osm_value` is "yes", which doesn't feel like something parallel to "restaurant" to me.

**Reply by mtmail** [reply]
There's dozens of possible values for `shop`, e.g. `shop=clothes` https://wiki.openstreetmap.org/wiki/Key:shop?uselang=en but `yes` is acceptable "When in doubt, you can also tag a shop as shop=yes" Examples https://www.openstreetmap.org/node/3587338993  https://www.openstreetmap.org/node/356074102  https://www.openstreetmap.org/node/345380438 It's probably similar with restaurants.
